src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,142 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
 
-                var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+            if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+            {
+                projectModel.AddProjectInformation(problemType: ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_NotPRProject, projectPath));
+                return;
+            }
 
-                // If the file was not found, print an error message and continue to next project
-                if (!File.Exists(assetsPath))
-                {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_AssetsFileNotFound,
-                        projectPath));
-                    Console.WriteLine();
-                }
-                else
+            var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+
+            if (!File.Exists(assetsPath))
+            {
+                projectModel.AddProjectInformation(ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_AssetsFileNotFound, projectPath));
+            }
+            else
+            {
+                var lockFileFormat = new LockFileFormat();
+                LockFile assetsFile = lockFileFormat.Read(assetsPath);
+
+                // Assets file validation
+                if (assetsFile.PackageSpec != null &&
+                    assetsFile.Targets != null &&
+                    assetsFile.Targets.Count != 0)
                 {
-                    var lockFileFormat = new LockFileFormat();
-                    var assetsFile = lockFileFormat.Read(assetsPath);
+                    // Get all the packages that are referenced in a project
+                    IEnumerable<FrameworkPackages> packages;
+                    try
+                    {
+                        packages = msBuild.GetResolvedVersions(project.FullPath, listPackageArgs.Frameworks, assetsFile, listPackageArgs.IncludeTransitive, includeProjects: listPackageArgs.ReportType == ReportType.Default);
+                    }
+                    catch (InvalidOperationException ex)
+                    {
+                        projectModel.AddProjectInformation(ProblemType.Error, ex.Message);
+                        return;
+                    }
+
 
-                    // Assets file validation
-                    if (assetsFile.PackageSpec != null &&
-                        assetsFile.Targets != null &&
-                        assetsFile.Targets.Count != 0)
+                    // If packages equals null, it means something wrong happened
+                    // with reading the packages and it was handled and message printed
+                    // in MSBuildAPIUtility function, but we need to move to the next project
+                    if (packages != null && packages.Any())
```

Comment:
The try/catch above returns so I don't imagine you need the null check anymore correct?

If you remove the null check you might be able to remove the comment too.

Reply:
Yes, removed it.